src/ActionParameterHandler.ts

```diff
@@ -96,11 +96,14 @@ export class ActionParameterHandler<T extends BaseDriver> {
   protected async normalizeParamValue(value: any, param: ParamMetadata): Promise<any> {
     if (value === null || value === undefined) return value;
 
-    // if param value is an object and param type match, normalize its string properties
-    if (
+    const isNormalisationNeeded =
       typeof value === 'object' &&
-      ['queries', 'headers', 'params', 'cookies'].some(paramType => paramType === param.type)
-    ) {
+      ['queries', 'headers', 'params', 'cookies'].some(paramType => paramType === param.type);
```

Comment:
`includes()` would be better in this case

Reply:
resolved https://github.com/typestack/routing-controllers/pull/708/files#diff-e388c22d7bdf11f58c29a3cf56e5fbf79de2efad1cbc8f0c54dc854cf146ca67R100